packages/core/src/db/models/Products.ts

```
@@ -329,6 +343,24 @@
 
       return product;
     }
+
+    public static async duplicateProduct(productId: string) {
+      const product = await models.Products.findOne({ _id: productId }).lean();
+
+      if (!product) throw new Error('Product not found');
+
+      const { _id, code, ...productData } = product;
+
+      const newCode = await this.generateCode();
```

Biome

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)